### ISSUE TITLE
Error during send notification

### ISSUE THREAD

**Issue by AlexFinn** [gogits/gogs#3856]
- Gogs version (or commit ref): 0.9.97.0901
- Git version: 1.9.1
- Operating system: ubuntu 14.04.5 LTS
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gogs.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log:
```
2016/11/01 08:13:55 [...les/mailer/mailer.go:201 processMailQueue()] [E] Fail to send emails [roman@example.com]: Subject: [setup] name (#7), issue comment - gomail: could not send email 1: gomail: invalid address "\"alx\" <username>": mail: missing @ in addr-spec
```

## Description
My mail settings:

```
[mailer]
ENABLED = true
HOST = email-smtp.us-east-1.amazonaws.com:587
FROM = git@example.com
USER = username
PASSWD = password
```
Why should smtp username have @ symbol?

**Comment by feroda** [comment]
I have a really similar problem:

`[...les/mailer/mailer.go:201 processMailQueue()] [E] Fail to send emails [devnull@example.com example@example.com]: Subject: [myrepo] Non inviare modifica TagAlarm a RabbitMQ o 
allo schema della scheda se db_initialized=False (#3), issue comment - gomail: could not send email 1: gomail: invalid address "\"Fabio\" <>": mail: invalid string`

**Comment by feroda** [comment]
bump?

**Comment by strk** [comment]
This is fixed in gitea, take a look: https://github.com/go-gitea/gitea

**Comment by einacio** [comment]
I'm having the same problem, where our smtp server needs the smt username in the form `user`, and not `user@domain.com`, which gogs expects.

I found these problems:
- some functions (as far as I could test, new user notification, register email, test mail, repo shared) uses mailer/mailer to add the message to the queue and it works ok, but pull request notification, and new push are using log/smtp, that doesn't use the FROM key of the ini, but instead is using the USER key and so it fails (since it's not a valid from-address) (the function with the problem is modules/log/smtp.go WriteMsg())
- when doing the install, it checks that username is an email, which is not neccesarily right

I tried to make the smtp module use SendAsync, but it gives me an "import cycle" error that I don't know how to resolve

(@strk, as far as i can see, gitea has the same bug in log/smtp)

**Comment by einacio** [comment]
i have replaced WriteMsg with a return nil but it still gives me the error :(

**Comment by strk** [comment]
Would you file an issue, or even better a pull request, to gitea? PRs are accepted fast there...

**Comment by Unknwon** [comment]
This is claimed to be fixed by merging #4055, please help test on `develop` branch.
